src/main.js

```diff
@@ -94,4 +97,42 @@ export const addListener = (selector, callback) => {
   return elements;
 };
 
+function debounce(f, t) {
+  return function (args) {
+    let previousCall = this.lastCall;
+    this.lastCall = Date.now();
+    if (previousCall && this.lastCall - previousCall <= t) {
+      clearTimeout(this.lastCallTimer);
+    }
+    this.lastCallTimer = setTimeout(() => f(args), t);
+  };
+}
+
+function onSearch(e) {
+  const searchQuery = e.target.value.trim();
+  const url = new URL(window.location.href);
+  url.searchParams.set('keyword', searchQuery);
```

Comment:
set page to 1, remove filter, name add filter as separate param with value name

Reply:
I agree. Fixed